shell/platform/darwin/ios/framework/Source/FlutterView.mm

```diff
@@ -77,8 +77,13 @@ - (instancetype)initWithDelegate:(id<FlutterViewEngineDelegate>)delegate
     // This line is necessary. CoreAnimation(or UIKit) may take this to do
     // something to compute the final frame presented on screen, if we don't set this,
     // it will make it take long time for us to take next CAMetalDrawable and will
-    // cause constant junk during rendering.
-    self.backgroundColor = UIColor.clearColor;
+    // cause constant junk during rendering. Note: setting any background color will
+    // disable opaque, which on full screen flutter apps removes the direct to display
+    // optimization. This can increase presentation time by a frame interval, which
+    // is problematic on high frame rate devices.
+    if (!opaque) {
```

Comment:
@luckysmg , I found that if we set the background color to any value, it resets the value of layer.opaque. If you have a fullscreen flutter view and opaque is true, then we get a direct to display optimization and the presentation delay decreases from ~21ms to ~14ms, which is particularly important for high frame rate devices.

See some findings in https://github.com/flutter/flutter/issues/134959

How are you using a FlutterView? Is it an add2app case where it is not the only view in the scene?

Reply:
Yes. We will change the layer's opaque to make the flutter view to be transparent and present it. The scene is that the new page is the transparent page, and it is presented on current page to do some thing like modal bottom sheet.

See https://github.com/alibaba/flutter_boost/blob/13eb9aed1dff9fcdfa6066666ba08a11b231af9f/ios/Classes/container/FBFlutterViewContainer.m#L137

Reply:
Also see https://github.com/alibaba/flutter_boost/blob/13eb9aed1dff9fcdfa6066666ba08a11b231af9f/ios/Classes/container/FBFlutterViewContainer.m#L221

Reply:
`If !opaque` and set a white color, will it make the `FlutterView` not transparent?

Reply:
ahh ok. Right now the problem is that we're unconditionally setting this clear color here. But setting the background color resets the value of opaque to NO, even if we requested it to be YES. So I think as long as your code is setting opaque = NO, this should keep working for you?

Reply:
Seems not working... Here init with delegate is too early to set bg color to white.

See
https://github.com/alibaba/flutter_boost/blob/13eb9aed1dff9fcdfa6066666ba08a11b231af9f/ios/Classes/container/FBFlutterViewContainer.m#L220

If we need the view is transparent, we will set the `opaque` to `NO` and don't set bg color(in above link's code)

Before this patch: the result FlutterView is will be not opaque and with clear color, so it will be transparent ✅
After this patch: the result FlutterView is will be not opaque and with white color, so it will not be transparent ❌

Reply:
So your code is starting out with opaque = YES, then you're setting it to opaque = NO later? I'm not really familiar enough with flutter_boost to follow the rendering paths there. I can change this so that if you later set opaque = NO, then we also set the background color to the clearColor?

Reply:
<  I can change this so that if you later set opaque = NO, then we also set the background color to the clearColor?

I think this can work. But I m not sure this is good or not... Because this has a implicit operation. Change `opaque`also change the bg color, maybe sometimes will confusing the developer..

Reply:
Yeah good point. I need to test if setting opaque _after_ we set the background color also resets the background color too. If not, maybe I just change the order and leave this as is.

Reply:
^_^